core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -76,6 +89,12 @@ private static boolean isAggregation(FunctionCall functionCall, Session session,
                 || functionCall.getOrderBy().isPresent();
     }
 
+    private static boolean isWindow(FunctionCall functionCall, Session session, Metadata metadata)
+    {
+        return metadata.isWindowFunction(session, functionCall.getName())
```

Comment:
Minor: If we need both of these checks, we probably want to call `functionCall.getWindow().isPresent()` first since the call through to `metadata.isWindowFunction` is almost certainly more expensive.

Reply:
That's a good callout. Will update it in the next revision.